internal/hash.h

```diff
@@ -164,6 +171,18 @@ RHASH_ST_TABLE(VALUE h)
 
 #endif
 
+static inline void
+RHASH_AR_TABLE_SET(VALUE h, struct ar_table_struct *tab)
+{
+    RHASH(h)->as.ar = tab;
+}
+
+static inline void
+RHASH_ST_TABLE_SET(VALUE h, st_table *tab)
+{
+    RHASH(h)->as.st = tab;
+}
```

Comment:
Why do we even have these? Can't we just put this in `hash_ar_table_set`/`rb_hash_st_table_set`?